### PR TITLE
Attributes for Rust

### PR DESCRIPTION
**Pull request by ropoctl**

### CODE REVIEW

**Comment by ropoctl** [comment]
@vitaut is there a servicelab that needs to be triggered for this?

**Comment by vitaut** [comment]
@ropoctl, sorry for a slow reply. Please let me know if you are still interested in having this reviewed and I'll find a POC from our Rust experts. You'll also need to rebase the PR.

File: thrift/annotation/rust.thrift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+include "thrift/annotation/scope.thrift"
+
+package "facebook.com/thrift/annotation/rust"
+
+@scope.Exception
+@scope.Field
+@scope.Struct
+struct Attributes{
+    1: list<string> attributes;
+} (thrift.uri = "facebook.com/thrift/annotation/rust/Attributes")
```
Comment (vitaut) [inline thrift/annotation/rust.thrift:26]: thrift.uri is redundant, please remove.

Comment (ropoctl) [reply thrift/annotation/rust.thrift]: Done

File: thrift/annotation/rust.thrift [hunk 2]
```diff
@@ -65,3 +65,10 @@ struct Adapter {
   // will use `fbthrift_adapters::DurationSecondsAdapter<i64>` as the adapter.
   1: string name;
 } (thrift.uri = "facebook.com/thrift/annotation/rust/Adapter")
+
+@scope.Exception
+@scope.Field
+@scope.Struct
+struct Attributes{
```
Comment (emersonford) [inline thrift/annotation/rust.thrift:72]: Can you change this to `@scope.Field`? If you want to apply a macro to a Thrift struct, you should use the `rust.derive = "..."` unstructured annotation (exists already) or the `@rust.Derives { derive = [...] }` structured annotation (will be putting up a PR for this shortly). 

Attribute macros allow you to read/modify the generated struct definition, which opens the door for a number of safety/compatibility issues. Derive macros allow only read-access to the struct definition (and subsequently enforce append-only behavior for macro generated code), which is much safer and should be the only allowed behavior for Rust Thrift.

**Comment by vitaut** [comment]
@ropoctl, do you plan to address feedback?

**Pull request closed without merging**